### ISSUE TITLE
SDK error on iOS 13+

### ISSUE THREAD

**Issue by sapradev** [aws/amazon-chime-sdk-ios#87]
Hi, i implemented Amazon Chime SDK in my react native project. It completely worked fine for iOS device of version 12. Also worked fine for simulator of version 13.4. But when i tried joining the meeting through iOS 13 or 13+ devices i got this error =>
Failed to start meeting, error: Error Domain=AmazonChimeSDK.MediaError Code=1 "(null)".
Using same code i was able to join meeting from iOS 12 device and iOS 13.4 simulator. But was not able to join from iOS 13 and 13+ devices.
![89641591-9f42b380-d8cf-11ea-8ce8-35cf15677388](https://user-images.githubusercontent.com/32849524/89756966-90434780-db01-11ea-8ca3-b8c0a33ee96b.png)

**Comment by zhinang-amazon** [comment]
Hi @sapradev this error means Audio did not start successfully. Can you provide what AmazonChimeSDK and AmazonChimeSDKMedia version are you using, and some logs when the issue happens?

**Comment by sapradev** [comment]
Hi, Thanks for replying.
i am using 
AmazonChimeSDK-0.8.1.tar.gz
AmazonChimeSDKMedia-0.6.0.tar.gz
with bitcode support

and i have attached a screenshot of a file which i took from demo project of react native chime. And at line number 210 the error is been generated as pop up on a iOS device 13.6.
As soon i click on start button it asks for microphone permission. User clicks ok. The screen where the user is shown he is joined to a meeting is displayed for a second and then it automatically pops this error
![Screenshot 2020-08-11 at 12 03 47 PM](https://user-images.githubusercontent.com/32849524/89865359-31e29b80-dbcb-11ea-93c6-7c4da06c8d7f.png)

**Comment by hokyungh** [comment]
Is there way to check whether `audioVideo.start` is called multiple times?

**Comment by sapradev** [comment]
Hi @hokyungh, thanks for replying
I guess no. 
While using 2 device i encountered a new issue or a bug. When i have already joined from 1st device and join from 2nd device and share video from 2nd device. the 1st device app gets crashed. On the go it gets crash. when i reopen the app and join the chime call i am able to see the video of 2nd device.
Please find the logs below.
![Screenshot 2020-08-14 at 1 27 31 PM](https://user-images.githubusercontent.com/32849524/90227294-ef6ec800-de31-11ea-8add-2c1dc1082072.png)

**Comment by sapradev** [comment]
Hi @hokyungh @zhinang-amazon,
It's been a while haven't heard from you guys.
i addressed a new issue as i mentioned earlier. While having 2 people connected to a same meeting. When one of them shares a video the app crashes for other person. I have attached the logs regarding the same which may help you and me to solve this issue.
Also i would like to update you on the version i am using. I have update my app to latest version of iOS SDK. Again reminding it's a react native application
![image](https://user-images.githubusercontent.com/32849524/90505791-d8e9a900-e170-11ea-99c0-839439d55f6a.png)
![image](https://user-images.githubusercontent.com/32849524/90505828-e6069800-e170-11ea-959f-acd444024518.png)

**Comment by hokyungh** [comment]
Sorry for late reply. Would you be able to attach whole .crash file? I believe you also used our React Native [demo](https://github.com/aws-samples/amazon-chime-react-native-demo/blob/master/src/components/RNVideoRenderView.js#L11) as reference right?

I don't think `Turn Cred Error logs` will crash the app, but we feel like we would need more information to reproduce this issue.

**Comment by sapradev** [comment]
Thanks for replying @hokyungh,
Yes i used your react native demo app to understand the flow and working.
I copied all the files like 
MeetingObserver both .h and .m
NativeMobileSDKBridge both .h and .m
and so on..
Also React Native files like meeting.js, login.js, etc. I have just changed some code according to my application.
Also as you said i have uploaded .crash file to my GDrive and shared the link with you below to download it.

https://drive.google.com/file/d/1wwROxwm2Iq3ANUhMijPF1QxbsWZDeFzt/view?usp=sharing

**Comment by sapradev** [comment]
Thanks for replying @hokyungh,
Yes, i did use your React Native demo app to understand the flow and code. Replicated all the .m and .h files as per instructions and also used files such as meeting.js, login.js, etc. Altered them according to application.
Also about .crash file sure i can share it. Just to be safe can you share me your email address so that i can share it with you.

**Comment by hokyungh** [comment]
Requested access

**Comment by sapradev** [comment]
Done @hokyungh,
Please have a look, will be waiting for your response

**Comment by hokyungh** [comment]
@sapradev Sorry for late reply. I was able to symbolicate the crash
```
partial apply for closure #1 in DefaultVideoTileController.onReceiveFrame(frame:videoId:attendeeId:pauseState:) (in AmazonChimeSDK) + 57
partial apply for thunk for @escaping @callee_guaranteed (@guaranteed VideoTileObserver) -> () (in AmazonChimeSDK) (<compiler-generated>:0)
specialized closure #1 in static ObserverUtils.forEach<A>(observers:observerFunction:) (in AmazonChimeSDK) + 408
partial apply for specialized  (in AmazonChimeSDK) + 27
thunk for @escaping @callee_guaranteed () -> () (in AmazonChimeSDK) (<compiler-generated>:0)
```

However, this doesn't seem to provide useful information for us. We retested with iPhone 11 simulator with latest SDK and media and still observed no crashes.

Would you be able to see if [code](https://github.com/aws-samples/amazon-chime-react-native-demo#how-to-bindunbind-the-video-tile) is what you have in your code?

Thanks! Let us know more details on how binding/unbinding logic is handled in your code.

**Comment by sapradev** [comment]
Hey @hokyungh,
Thanks a lot for helping out but it seems it still doesn't work for me.
Yes i have used the same code.
Find the screenshots below of binding and unbinding in NativeMobileSDKBridge.m file and RNVideo.js file

![Screenshot 2020-08-25 at 12 02 13 PM](https://user-images.githubusercontent.com/32849524/91163779-bcf38380-e6eb-11ea-8044-907f613b9946.png)
![Screenshot 2020-08-25 at 12 03 11 PM](https://user-images.githubusercontent.com/32849524/91163801-c5e45500-e6eb-11ea-8ccb-db530c4c7978.png)

Also find the crash video. It takes some time to crash after sharing a video from device. This time it gave me different error don't know why.
I have attached a video of my app crashing along with a screenshot of logs.
Also find crash file for the same on https://drive.google.com/file/d/1STXLGmYemcAeZ2bQASKjtMoJ5Nc23W0X/view?usp=sharing. Already shared with you so no need for permission. Find the zip file for logs & video below
[RN Chime.zip](https://github.com/aws/amazon-chime-sdk-ios/files/5123279/RN.Chime.zip)

**Comment by sapradev** [comment]
Hi @hokyungh ,
Also just to update. As soon as video is bind it automatically gets unbind and again tries to bind and then it crashes, i guess this is issue but have no clue how to resolve it. can you help out?
![Screenshot 2020-08-26 at 4 59 56 PM](https://user-images.githubusercontent.com/32849524/91298642-014f5400-e7be-11ea-84b7-463c7a0363d4.png)

**Comment by hokyungh** [comment]
@sapradev Do you mind providing me with exact details on repro steps?

1. Devices that are used in the testing
2. Exact repro step
3. If this is reproducible using demo app?

Following is the log I get from our demo application using 
0.8.1/0.6.0 and bitcode enabled. Tested with iPhone 6 Plus/IPhone 11 Simulator/Android Pixel 3XL.
```
2020-08-26 15:55:38.404189-0700 RNDemo[811:426016] [INFO] NativeMobileSDKBridge - Binding VideoView to Tile with tileId = 0
2020-08-26 15:55:38.404326-0700 RNDemo[811:426016] [INFO] NativeMobileSDKBridge - Unbinding the view from tile: tileId:  0, attendeeId: 18c130b9-b5f1-24db-a9cf-9d33149b7e8c
2020-08-26 15:55:38.404398-0700 RNDemo[811:426016] [INFO] NativeMobileSDKBridge - Binding the view to tile: tileId: 0, attendeeId: 18c130b9-b5f1-24db-a9cf-9d33149b7e8c
2020-08-26 15:55:41.429617-0700 RNDemo[811:426016] [INFO] NativeMobileSDKBridge - videoClientCameraSendIsAvailable
2020-08-26 15:56:23.383752-0700 RNDemo[811:426016] [INFO] NativeMobileSDKBridge - Attendee 18c130b9-b5f1-24db-a9cf-9d33149b7e8c volumeLevel changed to 1
2020-08-26 15:56:23.402671-0700 RNDemo[811:426016] [INFO] NativeMobileSDKBridge - Attendee 18c130b9-b5f1-24db-a9cf-9d33149b7e8c volumeLevel changed to 0
2020-08-26 15:56:24.162511-0700 RNDemo[811:426016] [INFO] NativeMobileSDKBridge - Attendee 18c130b9-b5f1-24db-a9cf-9d33149b7e8c volumeLevel changed to 1
2020-08-26 15:56:24.222523-0700 RNDemo[811:426016] [INFO] NativeMobileSDKBridge - Attendee 18c130b9-b5f1-24db-a9cf-9d33149b7e8c volumeLevel changed to 0
2020-08-26 15:56:24.983621-0700 RNDemo[811:426016] [INFO] NativeMobileSDKBridge - Attendee 18c130b9-b5f1-24db-a9cf-9d33149b7e8c volumeLevel changed to 2
2020-08-26 15:56:25.026496-0700 RNDemo[811:426016] [INFO] NativeMobileSDKBridge - Attendee 18c130b9-b5f1-24db-a9cf-9d33149b7e8c volumeLevel changed to 0
2020-08-26 15:56:25.783094-0700 RNDemo[811:426016] [INFO] NativeMobileSDKBridge - Attendee 18c130b9-b5f1-24db-a9cf-9d33149b7e8c volumeLevel changed to 2
2020-08-26 15:56:25.822414-0700 RNDemo[811:426016] [INFO] NativeMobileSDKBridge - Attendee 18c130b9-b5f1-24db-a9cf-9d33149b7e8c volumeLevel changed to 0
2020-08-26 15:56:26.583448-0700 RNDemo[811:426016] [INFO] NativeMobileSDKBridge - Attendee 18c130b9-b5f1-24db-a9cf-9d33149b7e8c volumeLevel changed to 2
2020-08-26 15:56:26.607563-0700 RNDemo[811:426016] [INFO] NativeMobileSDKBridge - Attendee 18c130b9-b5f1-24db-a9cf-9d33149b7e8c volumeLevel changed to 1
2020-08-26 15:56:26.663137-0700 RNDemo[811:426016] [INFO] NativeMobileSDKBridge - Attendee 18c130b9-b5f1-24db-a9cf-9d33149b7e8c volumeLevel changed to 0
2020-08-26 15:56:27.383452-0700 RNDemo[811:426016] [INFO] NativeMobileSDKBridge - Attendee 18c130b9-b5f1-24db-a9cf-9d33149b7e8c volumeLevel changed to 2
2020-08-26 15:56:27.403754-0700 RNDemo[811:426016] [INFO] NativeMobileSDKBridge - Attendee 18c130b9-b5f1-24db-a9cf-9d33149b7e8c volumeLevel changed to 1
2020-08-26 15:56:27.485528-0700 RNDemo[811:426016] [INFO] NativeMobileSDKBridge - Attendee 18c130b9-b5f1-24db-a9cf-9d33149b7e8c volumeLevel changed to 0
2020-08-26 15:56:27.683686-0700 RNDemo[811:426016] [INFO] NativeMobileSDKBridge - Attendee f631abdb#Ggggg mute
2020-08-26 15:56:27.683955-0700 RNDemo[811:426016] [INFO] NativeMobileSDKBridge - Attendee 18c130b9-b5f1-24db-a9cf-9d33149b7e8c volumeLevel changed to -1
2020-08-26 15:59:01.478306-0700 RNDemo[811:426016] [INFO] NativeMobileSDKBridge - Attendee 18c130b9-b5f1-24db-a9cf-9d33149b7e8c signalStrength changed to 1
2020-08-26 15:59:01.878574-0700 RNDemo[811:426016] [INFO] NativeMobileSDKBridge - Attendee 18c130b9-b5f1-24db-a9cf-9d33149b7e8c signalStrength changed to 2
2020-08-26 15:59:12.097586-0700 RNDemo[811:426016] [INFO] NativeMobileSDKBridge - Attendee b7c335d5-15c7-5534-b0bc-f5394fdc782a signalStrength changed to 1
2020-08-26 15:59:13.118021-0700 RNDemo[811:426016] [INFO] NativeMobileSDKBridge - Attendee b7c335d5-15c7-5534-b0bc-f5394fdc782a signalStrength changed to 2
2020-08-26 15:59:33.557065-0700 RNDemo[811:426678] [BoringSSL] nw_protocol_boringssl_get_output_frames(1301) [C5.2:2][0x10546a830] get output frames failed, state 8196
2020-08-26 15:59:33.562157-0700 RNDemo[811:426678] [BoringSSL] nw_protocol_boringssl_get_output_frames(1301) [C5.2:2][0x10546a830] get output frames failed, state 8196
2020-08-26 15:59:33.564469-0700 RNDemo[811:426678] TIC Read Status [5:0x0]: 1:57
2020-08-26 15:59:33.564544-0700 RNDemo[811:426678] TIC Read Status [5:0x0]: 1:57
2020-08-26 16:00:13.618195-0700 RNDemo[811:426016] [INFO] NativeMobileSDKBridge - AttendeeQuit(leave) : f4e80950#Dserwe
2020-08-26 16:00:13.621567-0700 RNDemo[811:426016] [INFO] NativeMobileSDKBridge - videoClientCameraSendIsAvailable
2020-08-26 16:00:15.890766-0700 RNDemo[811:426016] [INFO] NativeMobileSDKBridge - videoClientCameraSendIsAvailable
2020-08-26 16:00:46.898180-0700 RNDemo[811:426016] [INFO] NativeMobileSDKBridge - AttendeeQuit(drop) : 29f679bc#lkgkg
2020-08-26 16:03:27.960485-0700 RNDemo[811:426282] [INFO] NativeMobileSDKBridge - API/DefaultAudioVideoFacade/stop
2020-08-26 16:03:27.963316-0700 RNDemo[811:426282] [INFO] NativeMobileSDKBridge - Stopping VideoClient
2020-08-26 16:03:28.627690-0700 RNDemo[811:426016] [INFO] NativeMobileSDKBridge - videoClientDidStop
2020-08-26 16:03:28.628510-0700 RNDemo[811:426282] [INFO] NativeMobileSDKBridge - VideoClient is being destroyed
```

**Comment by sapradev** [comment]
Sorry for the delay @hokyungh,
The project is not based on Expo cli, it's based on React Native CLI. Though i implemented unimodules (Solution provided in link you gave above) but issue still persists.
1. Devices used are - IPad mini 2 (Device) iOS 12.4.5 and iPhone 11 (Simulator) iOS 13.4.1
I connect both of them to the same chime call and then start video from device and my simulator crash. Earlier before this i also tried the same using iPhone 6 and iPhone SE. Same issue when one shares video the app crashes for another one. Here iPhone SE is on latest iOS version and iPhone 6 is on iOS 12.

2. Repro steps - i just followed the steps exactly mentioned in [here](https://github.com/aws-samples/amazon-chime-react-native-demo)
i took all the files from that project into mine. Including all .m .h files and .js files. And on one of my button click i just initiate call. As per now audio works perfectly. there is no issue in audio. So i guess repro of project was perfect.
I updated my app to latest chime SDK with bitcode support.

**Comment by sapradev** [comment]
Hi @hokyungh,
Any updates? Repro steps needs anything additional for video? I shared details about repro already above. Audio works perfectly fine for me. Video behaves oddly.

**Comment by hokyungh** [comment]
Sorry for late reply @sapradev. Is there a way for us to reproduce this since you also mentioned that you cannot reproduce this on our demo application?

**Comment by sapradev** [comment]
Hi @hokyungh,
How can i help you do that? 
As what i did is just tried replicate your app. One thing is the screen where you accept the credentials and click on join meeting button is different on my side. On my side there is a button on a view. which on click runs your code (login one). All the data is filled from the backend so there is no front end with textfield like in yours. And there is no button either. Instead of your login screen i have redesigned it to a screen which just shows that its loading. then i switch to the call once everything is loaded. Audio tested on simulators as well as on devices. Works perfectly fine

**Comment by hokyungh** [comment]
Do you know if there is extra library being used? Any UI change on video?

**Comment by sapradev** [comment]
Hi @hokyungh,
Sorry for late reply.
No, i believe there are no extra libraries in entire project which points to video. Infact apart from video calling there is nothing regarding Video in application

**Comment by zhinang-amazon** [comment]
Hi @sapradev 

If you can consistently reproduce this issue, can you set breakpoint in `NativeMobileSDKBridge.m`  and make sure you are not invoking 
`self.meetingSession.audioVideo bindVideoViewWithVideoView:tileId:`
with nil?

In other words, check whether the view type casting `(RNVideoView*)view` and `[tileId integerValue]` are returning nil.

**Comment by zhinang-amazon** [comment]
Resolving since we have not heard back from requester.

**Comment by wuniencheng** [comment]
I faced this issue today. Similar to your case, I'm using React Native, and the error frequetly happened on iOS. In my case, my startMeeting() function call is put inside useEffect, like this:

```
useEffect(() => {
  if (callUUID && status === 'ON_CALL') {
    NativeFunction.startMeeting(meeting, attendee);
  }
}, [status]);
```

However, I've tried using startMeeting() elsewhere to join an existing meeting and it worked fine. I think the issue may happen because startMeeting() is called when Chime has not yet initialized. Therefore, I tried changing my code to this:

```
useEffect(() => {
  if (callUUID && status === 'ON_CALL') {
    setTimeout(() => {
      NativeFunction.startMeeting(meeting, attendee);
    }, 1000);
  }
}, [status]);
```

And the problem went away. It is just a simple workaround and might not suit your case, but it did solved my problem.
